cliquet/tests/test_utils.py

```diff
@@ -118,6 +119,26 @@ def test_current_service_returns_none_for_unexisting_patterns(self):
         self.assertEqual(current_service(request), None)
 
 
+def build_real_request(wsgi_environ):
+    from pyramid import request as pyramid_request
+    from pyramid import testing
+    from cliquet import includeme
+    from cliquet import DEFAULT_SETTINGS
+    config = testing.setUp(settings=DEFAULT_SETTINGS)
+    includeme(config)
+    request = pyramid_request.Request(wsgi_environ)
+    request.registry = config.registry
+    return request
+
+
+class BuildRequestTest(unittest.TestCase):
+
+    def test_built_request_has_cliquet_custom_methods(self):
+        original = build_real_request({'PATH_INFO': '/foo'})
+        request = build_request(original, {"path": "bar"})
+        self.assertIsNone(request.current_service)
```

Comment:
Ah, nevermind. I found it weird to have it assert that it returns None. Can be replaced with `assertTrue(hasattr(request, 'current_server'))` for ease of read.